Src/STRHAL_CAN.c

```diff
@@ -470,6 +482,11 @@ void FDCAN1_IT0_IRQHandler(void)
 			rec(id, rx_fifo->data.byte, l - 2);
 		FDCAN1->RXF1A = i & 0x7;
 	}
+	if (FDCAN1->IR & FDCAN_IR_BO) {

```

Comment:
It looks like this is triggered every time the bus-off state changes, not just when entering it:
```
Bit 19 BO: Bus-off status
0: Bus-off status unchanged
1: Bus-off status changed
```

But I don't think this causes a problem, right? Also reading `PSR` to check if we are currently in bus off state wouldn't be ideal because we would reset some of the "clear on read" bits.

Reply:
Good catch. It worked when I tested it but I can also add another check to see if the CCR Init Bit is set. That's how I did it in the Ecu_Lamarr.cpp implementation